grisera_api/observable_information/observable_information_service_mongodb.py

```diff
@@ -0,0 +1,282 @@
+        Args:
+            observable_information_id (int): Id of observable information
+        Returns:
+            Result of request as observable information object
+        """
+        observable_information = self.get_observable_information(
+            observable_information_id
+        )
+        if type(observable_information) is NotFoundByIdModel:
+            return NotFoundByIdModel(
+                id=observable_information_id,
+                errors={"errors": "observable information not found"},
+            )
+        return self.recording_service.remove_observable_information(
+            observable_information
+        )
+
+    def update_observable_information_relationships(
+        self,
+        observable_information_id: Union[str, int],
+        observable_information: BasicObservableInformationOut,
+    ):
+        """
+        Send request to graph api to update given observable information
+        Args:
+            observable_information_id (int): Id of observable information
+            observable_information (ObservableInformationIn): Relationships to update
+        Returns:
+            Result of request as observable information object
+        """
+        return self.recording_service.update_observable_information(
+            observable_information_id, observable_information.dict()
+        )
+
+    def _add_related_documents(
+        self,
+        observable_information: dict,
+        depth: int,
+        source: str,
+        recording: dict,
+    ):
+        """Recording is taken from previous get query"""
+        if depth > 0:
+            self._add_related_time_series(observable_information, depth, source)
+            self._add_related_modalities(observable_information, depth, source)
+            self._add_related_life_activities(observable_information, depth, source)
+            self._add_recording(observable_information, depth, source, recording)
+
+    def _add_recording(
+        self, observable_information: dict, depth: int, source: str, recording: dict
+    ):
+        """Recording has already added related documents"""
+        if source != "recording":
+            observable_information["recording"] = recording
+
+    def _add_related_modalities(
+        self, observable_information: dict, depth: int, source: str
+    ):
+        pass
+        # TODO add when modalities service is ready
+
+    def _add_related_life_activities(
+        self, observable_information: dict, depth: int, source: str
+    ):
+        pass
+        # TODO add when life_activities service is ready
+
+    def _add_related_time_series(
+        self, observable_information: dict, depth: int, source: str
+    ):
+        pass
+        # TODO add when time series service is ready
+
+    @staticmethod
+    def _get_recording_projection(query):
+        return {
+            Collections.OBSERVABLE_INFORMATION: {"$elemMatch": query},
+            "additional_properties": 1,
+            "participation_id": 1,
```

Comment:
why this ids are hardcoded? is it TODO?

Reply:
1s here are not IDs, in mongo projection "1" is the argument that specifies that field should be included in result.
[https://www.mongodb.com/docs/manual/tutorial/project-fields-from-query-results/](https://www.mongodb.com/docs/manual/tutorial/project-fields-from-query-results/)